openapi.go

```diff
@@ -457,6 +457,39 @@ func parseStructTags(t reflect.Type, schemaRef *openapi3.SchemaRef) {
 		propertyCopy := *property
 		propertyValue := *propertyCopy.Value
 
+		if field.Type.Kind() == reflect.Struct {
+			parseStructTags(field.Type, schemaRef.Value.Properties[jsonFieldName])
+		}
+
+		// basic Xml tag support
+		//
+		// We support the following xml tags:
+		// - "xml" field tag: specifies the XML element name
+		// - "xml=attr" field tag: specifies that the field is an XML attribute
+		// - "xml=wrapped" field tag: specifies that the field is wrapped in an XML element
+		xmlTag, ok := field.Tag.Lookup("xml")
+		if ok {
+			xmlTagName := strings.Split(xmlTag, ",")[0] // remove omitempty, etc
+			// if xml tag name is "-", don't add it to the schema
+			if xmlTagName != "-" {
```

Comment:
Instead of nesting more if can just do 

```
if xmlTagName == "-" {
	continue
}
```

Reply:
If we use `continue`, all subsequent code (such as handling examples, validation, and descriptions) will be skipped if the field isn't included in XML.  

In this section, only XML-related information is processed.

Reply:
I believe we would then need to change the JSON check above to handle the same case. I.E when the xml tag provide but not the json tag.

Reply:
Also I think it would make it a bit cleaner to perform both these actions in helpers so we can early return. 

We also will need to handle the case where both the `xml` and `json` tags are `-` we wouldn't want the description, examples, or validation processed. Kinda like how we do already with the continue in the `json` check for `-`. Would probably be worth to check that early for both so we can continue early.